Add type hints to urllib3.contrib._securetransport.low_level

hramezani opened the pull request:
<!---
Thanks for your contribution! ♥️

If this is your first PR to urllib3 please review the Contributing Guide:
https://urllib3.readthedocs.io/en/latest/contributing.html

Adhering to the Contributing Guide means we can review, merge, and release your change faster! :)
--->

sethmlarson commented on src/urllib3/contrib/_securetransport/low_level.py:
```diff
@@ -34,7 +46,7 @@ def _cf_data_from_bytes(bytestring):
 
 
-def _cf_dictionary_from_tuples(tuples):
+def _cf_dictionary_from_tuples(tuples: List[Tuple[Any, ...]]) -> CFDictionary:
```
Type should be `List[Tuple[Any, Any]]`

sethmlarson commented on src/urllib3/contrib/_securetransport/low_level.py:
```diff
@@ -123,10 +135,12 @@ def _cf_string_to_unicode(value):
     if string is not None:
         string = string.decode("utf-8")
-    return string
+    return string  # type: ignore
```
Can `string` somehow be `None` here?

sethmlarson commented on noxfile.py:
```diff
@@ -9,6 +9,7 @@
 # ignored.
 TYPED_FILES = {
     "src/urllib3/contrib/__init__.py",
+    "src/urllib3/contrib/_securetransport/low_level.py",
```
Forgot to mention it in the previous PR, can we add the `bindings.py` to this list? Just so we run mypy on it even if it's basically skipped.

sethmlarson reviewed (changes requested):
Looks pretty good, few comments then we can merge.

hramezani commented:
@sethmlarson Thanks for the review.
I addressed your comments related to type hints(force pushed to the first commit)
And added another commit to add `bindings.py` to `TYPED_FILES` list

sethmlarson reviewed (approved):
LGTM :rocket:
